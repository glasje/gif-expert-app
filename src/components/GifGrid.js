import React from 'react'
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from './hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">cargando...</p>}
            { <div className="card_grid animate__animated animate__fadeIn">


                {images.map((img) => (

                    <GifGridItem
                        key={img.id}
                        {...img} />


                ))}

            </div>}
        </>

    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}
