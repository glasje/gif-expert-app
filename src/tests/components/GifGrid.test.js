
import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../components/hooks/useFetchGifs';

jest.mock('../../components/hooks/useFetchGifs')
describe('Prueba en el <GifGrid/>', () => {

 
    const category = 'One Punch'
    test('debe de mostrar correctamente', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true })
        const wrapper = shallow(<GifGrid />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar item cuando se cargan imagenes', () => {
        const gifs = [{
            id: 'adv',
            url: 'jlkasdlkjdslkajjlkdas',
            title: 'sdffdssf'
        }];
        useFetchGifs.mockReturnValue({ data: gifs, loading: false })
        const wrapper = shallow(<GifGrid />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })

})
