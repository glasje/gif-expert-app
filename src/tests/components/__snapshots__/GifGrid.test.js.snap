// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el <GifGrid/> debe de mostrar correctamente 1`] = `
<Fragment>
  <h3 />
  <p
    className="animate__animated animate__flash"
  >
    cargando...
  </p>
  <div
    className="card_grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Prueba en el <GifGrid/> debe de mostrar item cuando se cargan imagenes 1`] = `
<Fragment>
  <h3 />
  <div
    className="card_grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="adv"
      key="adv"
      title="sdffdssf"
      url="jlkasdlkjdslkajjlkdas"
    />
  </div>
</Fragment>
`;
